packages/flutter_tts/tizen/src/flutter_tts_tizen_plugin.cc

```diff
@@ -20,124 +20,64 @@ class FlutterTtsTizenPlugin : public flutter::Plugin {
     registrar->AddPlugin(std::move(plugin));
   }
 
-  FlutterTtsTizenPlugin(flutter::PluginRegistrar *registrar)
-      : tts_(nullptr),
-        voice_type_(TTS_VOICE_TYPE_AUTO),
-        speed_(TTS_SPEED_AUTO),
-        speaking_(false),
-        await_speak_completion_(false) {
+  FlutterTtsTizenPlugin(flutter::PluginRegistrar *registrar) {
     channel_ =
         std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
             registrar->messenger(), "flutter_tts",
             &flutter::StandardMethodCodec::GetInstance());
     channel_->SetMethodCallHandler([this](const auto &call, auto result) {
       this->HandleMethodCall(call, std::move(result));
     });
-    EnsureTtsHandle();
-  }
 
-  virtual ~FlutterTtsTizenPlugin() {
-    if (tts_ != nullptr) {
-      tts_destroy(tts_);
-      tts_ = nullptr;
-    }
-  }
+    tts_ = std::make_unique<TextToSpeach>();
+    tts_->SetOnStateChanagedCallback(
+        [this](tts_state_e previous, tts_state_e current) -> void {
+          std::unique_ptr<flutter::EncodableValue> value =
+              std::make_unique<flutter::EncodableValue>(true);
+          if (current == TTS_STATE_PLAYING) {
+            if (previous == TTS_STATE_READY) {
+              channel_->InvokeMethod("speak.onStart", std::move(value));
+            } else if (previous == TTS_STATE_PAUSED) {
+              channel_->InvokeMethod("speak.onContinue", std::move(value));
+            }
+          } else if (current == TTS_STATE_PAUSED) {
+            channel_->InvokeMethod("speak.onPause", std::move(value));
+          } else if (current == TTS_STATE_READY) {
+            if (previous == TTS_STATE_PLAYING || previous == TTS_STATE_PAUSED) {
+              if (tts_->GetUttId()) {
```

Comment:
Hard to guess what this does at a glance but a nice solution!

Reply:
Thank you! I'll consider adding some comments or introducing a new method.